10up/ElasticPress
# Publicly expose a method for opening Instant Results.

### Description of the Change
Publicly expose a method for opening Instant Results, allowing third parties to open Instant Results manually.

I originally implemented this as a custom action hook, but I anticipate some refactoring for #2978 and didn't want to commit to an API just yet. For now this just makes `openModal()` publicly available.


Closes #2507

### How to test the Change
Calling the following code to a theme or plugin should cause Instant Results to open when the button is clicked:

```html
<button id="search-blocks" type="button">Search "Blocks"</button>

<script>
window.addEventListener('DOMContentLoaded', () => {
	document.getElementById('search-blocks').addEventListener('click', () => {
		window.epInstantResults.openModal({ search: "blocks" });
	});
});
</script>
```

### Changelog Entry
Added a method for developers to manually open Instant Results, `window.epInstantResults.openModal()`.

### Credits
Props @JakePT 

### Checklist:
